test/unit/cmap/message_stream.test.js

```diff
@@ -18,117 +20,116 @@ function bufferToStream(buffer) {
-          'hex'
-        ),
-        documents: [{ [LEGACY_HELLO_COMMAND]: 1 }]
-      },
-      {
-        description: 'valid OP_REPLY (partial)',
-        data: [
-          Buffer.from('37', 'hex'),
-          Buffer.from('0000', 'hex'),
-          Buffer.from(
-            '000100000001000000010000000000000000000000000000000000000001000000130000001069736d6173746572000100000000',
-            'hex'
-          )
-        ],
-        documents: [{ [LEGACY_HELLO_COMMAND]: 1 }]
-      },
-
-      {
-        description: 'valid OP_MSG',
-        data: Buffer.from(
-          '370000000100000000000000dd0700000000000000220000001069736d6173746572000100000002246462000600000061646d696e0000',
-          'hex'
-        ),
-        documents: [{ $db: 'admin', [LEGACY_HELLO_COMMAND]: 1 }]
-      },
-      {
-        description: 'valid multiple OP_MSG',
-        expectedMessageCount: 4,
-        data: Buffer.from(
-          '370000000100000000000000dd0700000000000000220000001069736d6173746572000100000002246462000600000061646d696e0000' +
-            '370000000100000000000000dd0700000000000000220000001069736d6173746572000100000002246462000600000061646d696e0000' +
-            '370000000100000000000000dd0700000000000000220000001069736d6173746572000100000002246462000600000061646d696e0000' +
-            '370000000100000000000000dd0700000000000000220000001069736d6173746572000100000002246462000600000061646d696e0000',
-          'hex'
-        ),
-        documents: [{ $db: 'admin', [LEGACY_HELLO_COMMAND]: 1 }]
-      },
-
-      {
-        description: 'Invalid message size (negative)',
-        data: Buffer.from('ffffffff', 'hex'),
-        error: 'Invalid message size: -1'
-      },
-      {
-        description: 'Invalid message size (exceeds maximum)',
-        data: Buffer.from('01000004', 'hex'),
-        error: 'Invalid message size: 67108865, max allowed: 67108864'
-      }
-    ].forEach(test => {
-      it(test.description, function (done) {
-        const error = test.error;
-        const expectedMessageCount = test.expectedMessageCount || 1;
-        const inputStream = bufferToStream(test.data);
-        const messageStream = new MessageStream();
+describe('MessageStream', function () {
+  context('when the stream is for a monitoring connection', function () {
+    const response = { isWritablePrimary: true };
+    let firstHello;
+    let secondHello;
+    let thirdHello;
+    let partial;
+
+    beforeEach(function () {
+      firstHello = generateOpMsgBuffer(response);
+      secondHello = generateOpMsgBuffer(response);
+      thirdHello = generateOpMsgBuffer(response);
+      partial = Buffer.alloc(5);
+      partial.writeInt32LE(100, 0);
+    });
 
-        let messageCount = 0;
-        messageStream.on('message', msg => {
-          messageCount++;
-          if (error) {
-            done(new Error(`expected error: ${error}`));
-            return;
-          }
+    it('only reads the last message in the buffer', async function () {
```

Comment:
Could we differentiate the content of the last message in some way so that we know that we in fact read the last message and not, say, the first?

Reply:
Your suggestion caught a bug. So thanks for that. :) https://github.com/mongodb/node-mongodb-native/pull/3256/commits/71fd458585c90c098e9690394ac572400b9c0c28